### PR TITLE
Add support for Redis Sentinels

### PR DESCRIPTION
**Pull request by romanalexander**

### CODE REVIEW

**Review by WiIIiam278 — CHANGES_REQUESTED** [review]
Interesting, but fair enough - happy to add this sort of thing if users are using it. One small request that'd also be neat for this PR: A config option to change the database No. (currently hardcoded to ID0)

File: common/src/main/java/net/william278/husksync/config/Settings.java [hunk 1]
```diff
@@ -134,6 +134,16 @@ public class Settings {
     @YamlKey("redis.use_ssl")
     private boolean redisUseSsl = false;
 
+    @YamlKey("redis.sentinel.master")
+    private String redisSentinelMaster = "";
```
[the remark this answers is not in the document]
Comment (romanalexander) [reply common/src/main/java/net/william278/husksync/config/Settings.java:139]: 👍

**Comment by romanalexander** [comment]
> Interesting, fair enough - happy to add this sort of thing if users are using it. Thanks for your work!
> 
> One small request that'd also be neat for this PR: A config option to change the database number (currently is hardcoded to 0)

Redis PubSub doesn't respect database numbers, so the channels would need to be changed to separate them out to not have strange cross-contamination issues. This could break existing plugins if they subscribe directly to the topics for behavior.

Also additionally some Redis topologies don't support database numbers which could cause confusion.

**Comment by WiIIiam278** [comment]
Interesting, understood - best left as is then :)

**Review by WiIIiam278 — APPROVED** [review]

**Pull request merged**